lib/wda/iproxy.js

```diff
@@ -1,98 +1,33 @@
-      `to device port ${this.deviceport} over USB for the device ${this.udid}`);
-    this.expectIProxyErrors = true;
-
-    return await new B((resolve, reject) => {
-      this.iproxy.on('exit', (code) => {
-        log.debug(`iproxy exited with code '${code}'`);
-        if (code) {
-          return reject(new Error(`iproxy exited with code '${code}'`));
-        }
-      });
-      this.iproxy.on('output', (stdout, stderr) => {
-        // do nothing if we expect errors
-        if (this.expectIProxyErrors) {
-          return;
-        }
-
-        let out = stdout || stderr;
-        for (let line of out.split('\n')) {
-          if (!line.length) {
-            continue;
-          }
-
-          if (line.includes('Resource temporarily unavailable')) {
-            // this generally happens when WDA does not respond,
-            // so print a more useful message
-            log.debug('Connection to WDA timed out');
-          } else {
-            iproxyLog.debug(line);
-          }
-        }
-      });
+  start () {
+    this.serverSocket = net.createServer(async (c) => {
+      try {
+        const socket = await utilities.connectPort(this.udid, this.deviceport);
+        socket.on('close', c.destroy);
+        c.on('close', socket.destroy);
+        c.pipe(socket);
+        socket.pipe(c);
+      } catch (e) {
+        c.destroy();
+      }
 
-      return (async () => {
-        try {
-          if ((await checkPortStatus(this.localport, '127.0.0.1')) === 'open') {
-            throw new Error(`The port #${this.localport} is occupied by an other app. ` +
-              `You can customize its value by setting the 'wdaLocalPort' capability.`);
-          }
-          await this.iproxy.start(0);
-          try {
-            await waitForCondition(async () => {
-              try {
-                return (await checkPortStatus(this.localport, '127.0.0.1')) === 'open';
-              } catch (ign) {
-                return false;
-              }
-            }, {
-              waitMs: IPROXY_STARTUP_TIMEOUT,
-              intervalMs: 300,
-            });
-            log.debug(`iProxy is running and is listening on port #${this.localport}`);
-          } catch (e) {
-            log.warn(`The local port ${this.localport} is still closed after ${IPROXY_STARTUP_TIMEOUT}ms. ` +
-              `Continuing anyway`);
-          }
-          resolve();
-        } catch (err) {
-          log.error(`Error starting iproxy: '${err.message}'`);
-          reject(new Error('Unable to start iproxy. Make sure libusbmuxd is installed and ' +
-            'PATH contains the folder, where the binary is located.'));
-        }
-      })();
     });
+    this.serverSocket.listen(this.localport);
   }
 
-  async quit () {
-    await killProcess('iproxy', this.iproxy);
-    this.expectIProxyErrors = true;
+  quit () {
+    if (!this.serverSocket) {
+      this.serverSocket.close();
```

Comment:
it makes sense to assign `this.serverSocket` to null after it was closed

Reply:
done